pkg/infrastructure/data_index.go

```diff
@@ -37,69 +32,11 @@ const (
-	log.Debugf("Querying KogitoApps in the namespace '%s' to inject Data Index Route ", namespace)
-
-	dcs, err := getKogitoAppsDCs(namespace, client)
-	if err != nil {
-		return err
-	}
-	log.Debugf("Found %s KogitoApps in the namespace '%s' ", len(dcs), namespace)
-	var dataIndexEndpoints ServiceEndpoints
-	if len(dcs) > 0 {
-		log.Debug("Querying Data Index route to inject into KogitoApps ")
-		dataIndexEndpoints, err = GetDataIndexEndpoints(client, namespace)
-		if err != nil {
-			return err
-		}
-		log.Debugf("Data Index route is '%s'", dataIndexEndpoints.HTTPRouteURI)
-	}
-
-	for _, dc := range dcs {
-		// here we compare the current value to avoid updating the app every time
-		if len(dc.Spec.Template.Spec.Containers) == 0 {
-			break
-		}
-		updateHTTP := framework.GetEnvVarFromContainer(dataIndexEndpoints.HTTPRouteEnv, &dc.Spec.Template.Spec.Containers[0]) != dataIndexEndpoints.HTTPRouteURI
-		updateWS := framework.GetEnvVarFromContainer(dataIndexEndpoints.WSRouteEnv, &dc.Spec.Template.Spec.Containers[0]) != dataIndexEndpoints.WSRouteURI
-		if updateHTTP {
-			log.Debugf("Updating dc '%s' to inject route %s ", dc.GetName(), dataIndexEndpoints.HTTPRouteURI)
-			framework.SetEnvVar(dataIndexEndpoints.HTTPRouteEnv, dataIndexEndpoints.HTTPRouteURI, &dc.Spec.Template.Spec.Containers[0])
-		}
-		if updateWS {
-			log.Debugf("Updating dc '%s' to inject route %s ", dc.GetName(), dataIndexEndpoints.WSRouteURI)
-			framework.SetEnvVar(dataIndexEndpoints.WSRouteEnv, dataIndexEndpoints.WSRouteURI, &dc.Spec.Template.Spec.Containers[0])
-		}
-		// update only once
-		if updateWS || updateHTTP {
-			if err := kubernetes.ResourceC(client).Update(&dc); err != nil {
-				return err
-			}
-		}
-	}
-	return nil
+	log.Debugf("Injecting Data-Index Route in kogito apps")
+	return injectURLIntoKogitoApps(client, namespace, dataIndexHTTPRouteEnv, dataIndexWSRouteEnv, &v1alpha1.KogitoDataIndexList{})
 }
 
 // GetDataIndexEndpoints queries for the Data Index URIs
 func GetDataIndexEndpoints(client *client.Client, namespace string) (dataIndexEndpoints ServiceEndpoints, err error) {
-	route := ""
-	dataIndexEndpoints = ServiceEndpoints{HTTPRouteEnv: dataIndexHTTPRouteEnv, WSRouteEnv: dataIndexWSRouteEnv}
-	route, err = getSingletonKogitoServiceRoute(client, namespace, &v1alpha1.KogitoDataIndexList{})
-	if err != nil {
-		return
-	}
-	if len(route) > 0 {
-		var routeURL *url.URL
-		routeURL, err = url.Parse(route)
-		if err != nil {
-			log.Warnf("Failed to parse data index route url (%s), set to empty: %s", route, err)
-			return
-		}
-		dataIndexEndpoints.HTTPRouteURI = routeURL.String()
-		if httpScheme == routeURL.Scheme {
-			dataIndexEndpoints.WSRouteURI = fmt.Sprintf("%s://%s", webSocketScheme, routeURL.Host)
-		} else {
-			dataIndexEndpoints.WSRouteURI = fmt.Sprintf("%s://%s", webSocketSecureScheme, routeURL.Host)
-		}
-	}
-	return
+	return getServiceEndpoints(client, namespace, dataIndexHTTPRouteEnv, dataIndexWSRouteEnv, &v1alpha1.KogitoDataIndexList{})
```

Comment:
Why did you move this function?

Reply:
I've refactored it so to reuse it in our trusty component, since we have to get/inject the trusty url as well. Would you suggest to do something different?

Reply:
Oh I see, you just move it to other file to be context aware.